controllers/volumereplication_controller.go

```diff
@@ -162,26 +174,49 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	case replicationv1alpha1.Primary:
 		err := r.markVolumeAsPrimary(volumeHandle, parameters, secret)
 		if err != nil {
+			_ = r.updateStatusStateAndMessage(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Secondary:
 		err := r.markVolumeAsSecondary(volumeHandle, parameters, secret)
 		if err != nil {
+			_ = r.updateStatusStateAndMessage(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	case replicationv1alpha1.Resync:
 		err := r.resyncVolume(volumeHandle, parameters, secret)
 		if err != nil {
+			_ = r.updateStatusStateAndMessage(instance, replicationv1alpha1.ReplicationFailure, err.Error())
 			return ctrl.Result{}, err
 		}
 
 	default:
-		return ctrl.Result{}, fmt.Errorf("unsupported image state %q", instance.Spec.ImageState)
+		err = fmt.Errorf("unsupported image state %q", instance.Spec.ImageState)
+		_ = r.updateStatusStateAndMessage(instance, replicationv1alpha1.ReplicationFailure, err.Error())
+		return ctrl.Result{}, err
+	}
+
+	instance.Status.LastCompletionTime = metav1.Now()
+	if instance.Spec.ImageState == replicationv1alpha1.Resync {
+		err = r.updateStatusStateAndMessage(instance, replicationv1alpha1.Replicating, "volume is marked for resyncing")
+	} else {
+		err = r.updateStatusStateAndMessage(instance, replicationv1alpha1.Replicating, "volume is marked "+string(instance.Spec.ImageState))
 	}
 
-	return ctrl.Result{}, nil
+	return ctrl.Result{}, err
+}
+
+func (r *VolumeReplicationReconciler) updateStatusStateAndMessage(instance *replicationv1alpha1.VolumeReplication, state replicationv1alpha1.ReplicationState, message string) error {
```

Comment:
You're always ignoring the error returned, so why return it?

Reply:
I am not always ignoring it. If you see the last call of the method in reconcile(). I am ignoring err at those places at which we going to return other err anyway.